src/components/implementation/tests/unit_schedtests/unit_schedlib.c

```diff
@@ -34,9 +34,9 @@ void
 test_thd_fn(void *data)
 {
 	while (1) {
-		int workiters = WORKITERS * ((int)data);
+		int workiters = WORKITERS * ((word_t)data);
```

Comment:
`data` here is a pointer type, so in x86_64 it will get casted into 32 bit if we use the `(int)` cast. Changes like this are mostly trying to remove warnings.

Reply:
Yeah, but it gets cast down to an int anyway, so I can't figure out why gcc complains about it. I guess it like the implicit cast down to int more than the explicit one? I'd hope it is the other way around. The cast relationship to warnings is still an area I have to understand more. Sigh.

Reply:
Yes, the compiler complains it as an implicit cast down. So do we still need to keep it as a `word_t`?

Reply:
In your updated code it is doing an implicit cast from `word_t` down to `int`. Your last comments says that the compiler complains about the implicit cast. The implication is that *with* your change, there is a warning, right?

I don't understand why the compiler wouldn't complain about your code's implicit cast, but would complain about the explicit cast in the previous code.

Reply:
`warning: cast from pointer to integer of different size [-Wpointer-to-int-cast]`
Tested again, my memory was wrong. The compiler complains a cast down of different size. But as you can see, there is another implicit cast down from the `WORKITERS * ((word_t)data)` to a `int`.

Reply:
warning: cast from pointer to integer of different size [-Wpointer-to-int-cast]
Tested again, my memory was a little bit wrong. The compiler complains a cast down of different size. But as you can see, there is another implicit cast down from the WORKITERS * ((word_t)data) to a int.

Reply:
`warning: cast from pointer to integer of different size [-Wpointer-to-int-cast]`
Tested again, my memory was a little bit wrong. The compiler complains a cast down of different size. But as you can see, there is another implicit cast down from the `WORKITERS * ((word_t)data)` to a `int`.